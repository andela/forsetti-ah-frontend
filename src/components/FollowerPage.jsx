import React, { Component } from 'react';
import {
  TabContent,
  Nav,
  NavItem,
  NavLink
} from 'reactstrap';
import { FollowerPaneComponent } from './FollowerPane';
import { FollowingPaneComponent } from './FollowingPane';

class FollowerPage extends Component {
  constructor(props) {
    super(props);

    this.toggle = this.toggle.bind(this);
    this.state = {
      activeTab: '1'
    };
  }

  toggle(tab) {
    const { activeTab } = this.state;
    if (activeTab !== tab) {
      this.setState({
        activeTab: tab
      });
    }
  }

  render() {
    const { activeTab } = this.state;
    return (
      <div className='container m-5 mx-auto'>
        <Nav tabs>
          <NavItem className='w-50'>
            <NavLink
              onClick={() => { this.toggle('1'); }}
              className={`text-center text-uppercase font-weight-bold ${activeTab === '1' ? 'tab-active-class' : ''}`}
            >
              Followers
            </NavLink>
          </NavItem>
          <NavItem className='w-50'>
            <NavLink
              onClick={() => { this.toggle('2'); }}
              className={`text-center text-uppercase font-weight-bold ${activeTab === '2' ? 'tab-active-class' : ''}`}
            >
              Followings
            </NavLink>
          </NavItem>
        </Nav>
        <TabContent activeTab={activeTab}>
          <FollowerPaneComponent />
          <FollowingPaneComponent />
        </TabContent>
      </div>
    );
  }
}

export default FollowerPage;
