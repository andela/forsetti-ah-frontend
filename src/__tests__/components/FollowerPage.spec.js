import React from 'react';
import { shallow } from 'enzyme';
import FollowerPage from '../../components/FollowerPage';

describe('Follower Page', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<FollowerPage />);
    expect(wrapper).toMatchSnapshot();
  })
});
