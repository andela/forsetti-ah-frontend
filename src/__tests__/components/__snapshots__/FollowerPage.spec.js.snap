// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Follower Page should match snapshot 1`] = `ShallowWrapper {}`;
